app/src/main/java/org/mifos/mobile/ui/savings_account_application/SavingsAccountApplicationFragment.kt

```diff
@@ -0,0 +1,145 @@
+import org.mifos.mobile.models.accounts.savings.SavingsAccountApplicationPayload
+import org.mifos.mobile.models.accounts.savings.SavingsAccountUpdatePayload
+import org.mifos.mobile.models.accounts.savings.SavingsWithAssociations
+import org.mifos.mobile.ui.activities.base.BaseActivity
+import org.mifos.mobile.ui.enums.SavingsAccountState
+import org.mifos.mobile.ui.fragments.base.BaseFragment
+import org.mifos.mobile.utils.Constants
+import org.mifos.mobile.utils.DateHelper
+import org.mifos.mobile.utils.Network
+import org.mifos.mobile.utils.ParcelableAndSerializableUtils.getCheckedParcelable
+import org.mifos.mobile.utils.ParcelableAndSerializableUtils.getCheckedSerializable
+import org.mifos.mobile.utils.getTodayFormatted
+import javax.inject.Inject
+
+/*
+* Created by saksham on 30/June/2018
+*/
+@AndroidEntryPoint
+class SavingsAccountApplicationFragment : BaseFragment() {
+
+    @Inject
+    lateinit var preferencesHelper: PreferencesHelper
+
+    private val viewModel: SavingsAccountApplicationViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (activity as? BaseActivity)?.hideToolbar()
+        if (arguments != null) {
+            viewModel.setSavingsAccountState(
+                arguments?.getCheckedSerializable(
+                    SavingsAccountState::class.java, Constants.SAVINGS_ACCOUNT_STATE
+                ) as SavingsAccountState
+            )
+            viewModel.setSavingsWithAssociations(
+                arguments?.getCheckedParcelable(
+                    SavingsWithAssociations::class.java, Constants.SAVINGS_ACCOUNTS
+                )
+            )
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        viewModel.loadSavingsAccountApplicationTemplate(preferencesHelper.clientId, viewModel.savingsAccountState)
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                MifosMobileTheme {
+                    SavingsAccountApplicationScreen(
+                        uiState = viewModel.savingsAccountApplicationUiState.value,
+                        savingsWithAssociations = viewModel.savingsWithAssociations,
+                        navigateBack = { activity?.finish() },
+                        retryConnection = { onRetry() },
+                        submit = { productId, clientId ->
+                            onSubmit(
+                                productId = productId,
+                                clientId = clientId
+                            )
+                        }
+                    )
+                }
+            }
+        }
+    }
+
+    private fun submitSavingsAccountApplication(productId: Int, clientId: Int) {
```

Comment:
move this to viewmodel and add callback for toast so you can have access to context in compose and show toast.

submitSavingsAccountApplication(productId: Int, clientId: Int, showToast: (String) -> Unit)
now you can directly call this from view model to composable